x-pack/plugins/profiling/server/index.ts

```diff
@@ -9,8 +9,23 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import type { PluginConfigDescriptor, PluginInitializerContext } from '@kbn/core/server';
 import { ProfilingPlugin } from './plugin';
 
+const commonConfigSchema = schema.object({
+  host: schema.string(),
+  secret_token: schema.maybe(schema.string()),
+  tls_enabled: schema.boolean(),
+  tls_supported_protocols: schema.arrayOf(schema.string()),
+  tls_certificate_path: schema.string(),
+  tls_key_path: schema.string(),
+});
+
 const configSchema = schema.object({
   enabled: schema.boolean({ defaultValue: false }),
+  config: schema.maybe(
+    schema.object({
+      simbolizer: commonConfigSchema,
+      collector: commonConfigSchema,
+    })
+  ),
```

Comment:
With the current schema for `ProfilingConfig`, `config/kibana[.dev].yml` would look like this:

```
xpack.profiling.enabled: true

xpack.profiling.config.collector.host: ""
xpack.profiling.config.collector.secret_token: ""
xpack.profiling.config.collector.tls_enabled: true
xpack.profiling.config.collector.tls_supported_protocols: []
xpack.profiling.config.collector.tls_certificate_path: ""
xpack.profiling.config.collector.tls_key_path: ""

xpack.profiling.config.symbolizer.host: ""
xpack.profiling.config.symbolizer.secret_token: ""
xpack.profiling.config.symbolizer.tls_enabled: true
xpack.profiling.config.symbolizer.tls_supported_protocols: []
xpack.profiling.config.symbolizer.tls_certificate_path: ""
xpack.profiling.config.symbolizer.tls_key_path: ""

xpack.profiling.elasticsearch.hosts: "url"
xpack.profiling.elasticsearch.username: "string"
xpack.profiling.elasticsearch.password: "string"
```

It may be clearer and/or simpler that the configuration looks like this:

```
xpack.profiling.enabled: true

xpack.profiling.collector.host: ""
xpack.profiling.collector.secret_token: ""
xpack.profiling.collector.tls_enabled: true
xpack.profiling.collector.tls_supported_protocols: []
xpack.profiling.collector.tls_certificate_path: ""
xpack.profiling.collector.tls_key_path: ""

xpack.profiling.symbolizer.host: ""
xpack.profiling.symbolizer.secret_token: ""
xpack.profiling.symbolizer.tls_enabled: true
xpack.profiling.symbolizer.tls_supported_protocols: []
xpack.profiling.symbolizer.tls_certificate_path: ""
xpack.profiling.symbolizer.tls_key_path: ""

xpack.profiling.elasticsearch.hosts: "url"
xpack.profiling.elasticsearch.username: "string"
xpack.profiling.elasticsearch.password: "string"
```

This would mean `configSchema` could be update to:

```suggestion
const configSchema = schema.object({
  enabled: schema.boolean({ defaultValue: false }),
  symbolizer: schema.maybe(commonConfigSchema),
  collector: schema.maybe(commonConfigSchema),
```

I would also suggest renaming `ConfigType` and `commonConfigSchema` to something like `PackageInputType` and `packageInputSchema`, respectively. My intent is to convey that the common schema for the symbolizer and collector is related to the input variable for those packages, so any similar naming scheme would work also.

Reply:
that looks great, thanks for the suggestions!